File: cpp/alerts/scrape_failed.cpp
#include "./scrape_failed.h"
#include "cpp/error.h"
#include "rust/cxx.h"

namespace ltrs {

rust::Str scrape_failed_alert_get_error_message(lt::scrape_failed_alert* alert) {
    return alert->error_message();
}

Error scrape_failed_alert_get_error(lt::scrape_failed_alert* alert) {
    return error_code_to_error(alert->error);
}

uint8_t scrape_failed_alert_get_version(lt::scrape_failed_alert* alert) {
    return static_cast<uint8_t>(alert->version);
}

} // namespace ltrs

File: src/ffi/alerts/scrape_failed.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type scrape_failed_alert = crate::ffi::ffi::scrape_failed_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/scrape_failed.h");
        type Error = crate::ffi::error::ffi::Error;
        unsafe fn scrape_failed_alert_get_error_message<'a>(
            alert: *mut scrape_failed_alert,
        ) -> &'a str;
        unsafe fn scrape_failed_alert_get_error(alert: *mut scrape_failed_alert) -> Error;
        unsafe fn scrape_failed_alert_get_version(alert: *mut scrape_failed_alert) -> u8;
    }
}
